### PR TITLE
AR-191 Make artifact jobs all work for Ocata

### PR DESCRIPTION
**Pull request by odyssey4me**
Unfortunately all the artifact jobs were giving a false pass due to a mistake in the job config. This has now been resolved so they're testing properly now, but all the artifact builds are broken. This PR will be a series of patches to fix them again.

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_all

File: scripts/artifacts-building/git/build-git-artifacts.sh [hunk 1]
```diff
@@ -41,7 +41,13 @@ export ANSIBLE_ROLE_FETCH_MODE="git-clone"
 ## Main ----------------------------------------------------------------------
 
 # Bootstrap Ansible
-./scripts/bootstrap-ansible.sh
+# This script is sourced to ensure that the common
+# functions and vars are available.
+source scripts/bootstrap-ansible.sh
+
+# Now use GROUP_VARS of OSA and RPC
+sed -i "s|GROUP_VARS_PATH=.*|GROUP_VARS_PATH=\"\${GROUP_VARS_PATH:-${OA_DIR}/playbooks/inventory/group_vars/:${BASE_DIR}/group_vars/:/etc/openstack_deploy/group_vars/}\"|" /usr/local/bin/openstack-ansibl
```
Comment (evrardjp) [inline scripts/artifacts-building/git/build-git-artifacts.sh]: little pasting problem?

File: scripts/artifacts-building/apt/build-apt-artifacts.sh [hunk 2]
```diff
@@ -87,14 +87,15 @@ set -x
 # Ensure that the repo server public key is a known host
 grep "${REPO_HOST}" ~/.ssh/known_hosts || echo "${REPO_HOST} $(cat $REPO_HOST_PUBKEY)" >> ~/.ssh/known_hosts
 
-#Append host to [mirrors] group
+# Create an inventory to use
 echo '[mirrors]' > /opt/inventory
 echo "repo ansible_host=${REPO_HOST} ansible_user=${REPO_USER} ansible_ssh_private_key_file='${REPO_KEYFILE}' " >> /opt/inventory
+echo "localhost ansible_python_interpreter: '/usr/bin/python2'" >> /opt/inventory
```
Comment (evrardjp) [inline scripts/artifacts-building/apt/build-apt-artifacts.sh]: I think for static inventories, it's probably better to use the format ansible_python_interpreter='/usr/bin/python2'

Comment (odyssey4me) [inline scripts/artifacts-building/apt/build-apt-artifacts.sh]: of course, thanks

File: scripts/artifacts-building/containers/patches/repo_server [hunk 3]
```diff
@@ -20,3 +20,6 @@ index 7fb1949..34cb748 100644
    tags:
      - pkg-repo-user
      - repo-key
+-- 
+2.11.0 (Apple Git-81)
```
Comment (evrardjp) [inline scripts/artifacts-building/containers/patches/repo_server]: What is this?

File: scripts/artifacts-building/containers/patches/os_keystone [hunk 4]
```diff
@@ -1,30 +1,15 @@
...
--    - Restart service
- 
- - name: Install pip packages
-   pip:
 diff --git a/tasks/keystone_pre_install.yml b/tasks/keystone_pre_install.yml
-index fbc668b..553c004 100644
+index 8a8a85c..c348d46 100644
 --- a/tasks/keystone_pre_install.yml
 +++ b/tasks/keystone_pre_install.yml
 @@ -46,7 +46,6 @@
```
Comment (evrardjp) [inline scripts/artifacts-building/containers/patches/os_keystone]: Here is what I meant with line numbers for patches applying well.

File: scripts/artifacts-building/containers/patches/repo_server (hunk 3 above)
Comment (odyssey4me) [inline scripts/artifacts-building/containers/patches/repo_server]: It's output from the fresh patch I put together. It replaces whatever your patch was previously.

File: scripts/artifacts-building/containers/patches/os_keystone (hunk 4 above)
Comment (odyssey4me) [inline scripts/artifacts-building/containers/patches/os_keystone]: I made completely fresh patches, and tested them.

File: scripts/artifacts-building/containers/patches/repo_server (hunk 3 above)
Comment (evrardjp) [inline scripts/artifacts-building/containers/patches/repo_server]: I am not sure these two lines are standard.

**Comment by odyssey4me** [comment]
recheck cit_artifact-xenial-Container

**Comment by odyssey4me** [comment]
recheck_cit_artifact-xenial-Container

**Comment by odyssey4me** [comment]
recheck_cit_artifact_xenial_Container

**Comment by evrardjp** [comment]
I'm fine for it when it works.

**Comment by odyssey4me** [comment]
I'm just working out the final issues with the container artifact builds, then this will be ready for review.

**Comment by odyssey4me** [comment]
@alextricity25 The changes here are explicitly to cater for a branch that has no artifacts yet, so the master branch is perfect. Once I have it working I'll test against a branch that *does* have artifacts too to verify that it's safe.

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by hughsaunders** [comment]
recheck_cit_xenial_deploy_swift

**Comment by odyssey4me** [comment]
recheck_cit_artifact_xenial_Container

**Comment by odyssey4me** [comment]
This patch is now ready for review.

File: scripts/artifacts-building/containers/build-process.sh [hunk 5]
```diff
@@ -89,20 +93,60 @@ git config --global user.email "rcbops@rackspace.com"
...
+# right artifacts to use. To ensure that we can still do a PR test
+# when there are no python artifacts, we need to override a few
+# things.
+if ! python_artifacts_available; then
+    # As there are no wheels available for this release, we will
+    # need to enable developer_mode for the role install.
+    echo "developer_mode: yes" >> ${OA_OVERRIDES}
+
+    # As there are is not pre-build constraints file available
+    # we will need to use those from upstream.
```
Comment (d34dh0r53) [inline scripts/artifacts-building/containers/build-process.sh]: Nit: Should be "As there are not pre-built constraints files available we will need to use those from upstream."

**Pull request merged**